code/modules/reagents/reagent_containers/food/snacks.dm

```diff
@@ -12,19 +12,67 @@
 	var/dried_type = null
 	var/dry = 0
 	var/dryness = 0 //Used by drying rack. Represents progress towards Dry state
+	var/junk_food = FALSE
 	var/nutriment_amt = 0
 	var/list/nutriment_desc = list("food" = 1)
 	center_of_mass = list("x"=16, "y"=16)
 	w_class = ITEM_SIZE_SMALL
 
-	var/sanity_gain = 0.2 //Per bite
+	var/sanity_gain = 0.2 //per nutriment
 	var/cooked = FALSE
 
 /obj/item/weapon/reagent_containers/food/snacks/Initialize()
 	. = ..()
 	if(nutriment_amt)
 		reagents.add_reagent("nutriment", nutriment_amt, nutriment_desc)
 
+/obj/item/weapon/reagent_containers/food/snacks/proc/get_sanity_gain(mob/living/carbon/eater) //sanity_gain Per bite
+	var/current_nutriment = reagents.get_reagent_amount("nutriment")
+	var/nutriment_percent = current_nutriment/reagents.total_volume
+	var/nutriment_eaten = min(reagents.total_volume, bitesize) * nutriment_percent
+	var/base_sanity_gain_pb = nutriment_eaten * sanity_gain
+	var/message
+	if(!iscarbon(eater))
+		return  list(0, message)
+	if(eater.nutrition > eater.max_nutrition*0.95)
+		message = "You are satisfied and you don't need to eat more."
+		return  list(0, message)
+	if(!base_sanity_gain_pb)
+		message = "This food does not help calm your nerves."
+		return  list(0, message)
+	var/sanity_gain_pb = base_sanity_gain_pb
+	message = "Food helps you relax."
+	if(cooked)
+		sanity_gain_pb += base_sanity_gain_pb * 0.2
+	if(junk_food || !cooked)
+		return  list(sanity_gain_pb, message)
+	var/table = FALSE
+	var/companions = FALSE
+	var/view_death = FALSE
+	for(var/C in circleview(eater, 3))
+		if(istype(C, /obj/structure/table))
+			if(!in_range(C, eater) || table)
+				continue
+			table = TRUE
+			message += " Eating is more comfortable using a table."
+			sanity_gain_pb += base_sanity_gain_pb * 0.1
+
+		else if(ishuman(C))
+			var/mob/living/carbon/human/H = C
+			if(H == eater)
+				continue
+			if(H.is_dead())
+				view_death = TRUE
+			companions = TRUE
+	if(companions)
+		sanity_gain_pb += base_sanity_gain_pb * 0.3
+		message += " The company make the food taste much better."
```

Comment:
> " The company _makes_ - "
Mind the spaces at the beginning as well.

Reply:
That's intentionally there, is there a better way to do it?

Reply:
I just noticed the `+=`, sorry about that.